mlrun/db/httpdb.py

```diff
@@ -2770,6 +2770,66 @@ def trigger_migrations(self) -> Optional[schemas.BackgroundTask]:
             )
         return None
 
+    def get_projects_dataschemas(self):
+        """Provide data schemas (feature store schemas) of MLRun projects as collection.
+
+        Examples::
+
+            ...
+            # connection to solution
+            db = mlrun.get_run_db()
+
+            # print data schema of all projects in JSON format
+            print(json.dumps(db.get_projects_dataschemas(),
+                ensure_ascii = False,
+                indent=2,
+                allow_nan=True)
+            )
+
+        :return:    Collection of data schemas (feature store schemas)
+        """
+
+        dataschema = []
+
+        # iteration via projects
+        for project in self.list_projects():
+            project_data_schema = {}
+            project_data_schema["project_name"] = project.name
+            if project.spec.description:
+                project_data_schema["project_description"] = project.spec.description
+            dataschema.append(project_data_schema)
+
+            # iteration via feature sets
+            feature_sets = self.list_feature_sets(project.name)
+            if feature_sets:
+                project_feature_sets = []
+                project_data_schema["feature_sets"] = project_feature_sets
+                for feature_set in feature_sets:
+                    if feature_set:
+                        featureset_data_schema = {}
+                        featureset_data_schema["name"] = feature_set.fullname
+                        if feature_set.spec.description:
+                            featureset_data_schema[
+                                "description"
+                            ] = feature_set.spec.description
+                        project_feature_sets.append(featureset_data_schema)
+
+                        # iteration via entities
+                        if feature_set.spec.entities:
+                            featureset_data_schema["entities"] = [
+                                entity.to_dict(["name", "description", "value_type"])
+                                for entity in feature_set.spec.entities
+                            ]
+
+                        # iteration via features
+                        if feature_set.spec.features:
+                            featureset_data_schema["features"] = [
+                                feature.to_dict(["name", "description", "value_type"])
+                                for feature in feature_set.spec.features
+                            ]
```

Comment:
same here, it will reduce the need for more indentations and will make the code more cleaner.